src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts

```diff
@@ -0,0 +1,172 @@
+import { CaseDate, CaseWithId } from '../../../app/case/case';
+import {
+  CurrentOrPreviousProceedings,
+  Document,
+  OrderDocumentInfo,
+  OtherProceedingDetails,
+  ProceedingDetailsData,
+  Proceedings,
+  ProceedingsOrderDataInterface,
+  ProceedingsOrderInterface,
+  ProceedingsOrderTypeInterface,
+  ProceedingsOrderTypes,
+  Respondent,
+} from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
+
+import { ANYTYPE } from './dateformatter';
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const setProceedingDetails = (UserCase, respondent: Respondent, req: AppRequest): Respondent => {
```

Comment:
This method is creating side effects.

Reply:
made changes and used local variable as discussed